c10/metal/utils.h

```diff
@@ -85,6 +85,45 @@ struct OpMathType<bfloat> {
   using type = float;
 };
 #endif
+
+// Type promotion structure for higher precision accumulation
+template <typename T>
+struct AccumulationType {
```

Comment:
There are already `opmath_t`, which is exactly the same (plus have logic for integers)

Reply:
You've convinced me, but do you mind keeping it consistent with `opmath_t` (i.e. just being a type alias and no cast methods, as they are trivial and don't need to be in the header files, are they?)

Reply:
done!